geoapi/tasks/external_data.py

```diff
@@ -229,10 +235,12 @@ def import_from_agave(userId: int, systemId: str, path: str, projectId: int):
                     continue
 
                 # If its a RApp project folder, grab the metadata from tapis meta service
-                if Path(item_system_path).match("*/RApp/*"):
-                    logger.info("RApp import {path}".format(path=item_system_path))
+                if is_member_of_rapp_project_folder(item_system_path):
+                    logger.info("RApp: importing:{} for user:{}".format(item_system_path, user.username))
+                    if item.path.suffix.lower().lstrip('.') not in FeaturesService.ALLOWED_GEOSPATIAL_FEATURE_ASSET_EXTENSIONS:
+                        logger.info("{path} is unsupported; skipping.".format(path=item_system_path))
+                        continue
                     listing = client.listing(systemId, item.path)[0]
-                    # f = client.getFile(systemId, item.path)
                     meta = client.getMetaAssociated(listing.uuid)
```

Comment:
@reptillicus 
So *each* file in a RApp folder has metadata (with geolocation) on it?

I don't have a project reated by a Rapp to test this.

Reply:
Figured out what was the serivce-account needed to be used (Des-1853)